Content.Shared/Weapons/Melee/MeleeThrowOnHitSystem.cs

```diff
@@ -15,113 +16,72 @@ namespace Content.Shared.Weapons.Melee;
 /// </summary>
 public sealed class MeleeThrowOnHitSystem : EntitySystem
 {
-    [Dependency] private readonly IGameTiming _timing = default!;
     [Dependency] private readonly SharedTransformSystem _transform = default!;
     [Dependency] private readonly SharedPhysicsSystem _physics = default!;
-
+    [Dependency] private readonly UseDelaySystem _delay = default!;
+    [Dependency] private readonly SharedStunSystem _stun = default!;
+    [Dependency] private readonly ThrowingSystem _throwing = default!;
     /// <inheritdoc/>
     public override void Initialize()
     {
         SubscribeLocalEvent<MeleeThrowOnHitComponent, MeleeHitEvent>(OnMeleeHit);
-        SubscribeLocalEvent<MeleeThrownComponent, ComponentStartup>(OnThrownStartup);
-        SubscribeLocalEvent<MeleeThrownComponent, ComponentShutdown>(OnThrownShutdown);
-        SubscribeLocalEvent<MeleeThrownComponent, StartCollideEvent>(OnStartCollide);
+        SubscribeLocalEvent<MeleeThrowOnHitComponent, ThrowDoHitEvent>(OnThrowHit);
     }
 
-    private void OnMeleeHit(Entity<MeleeThrowOnHitComponent> ent, ref MeleeHitEvent args)
+    private void OnMeleeHit(Entity<MeleeThrowOnHitComponent> weapon, ref MeleeHitEvent args)
     {
-        var (_, comp) = ent;
+        // TODO: MeleeHitEvent is weird. Why is this even raised if we don't hit something?
         if (!args.IsHit)
             return;
 
-        var mapPos = _transform.GetMapCoordinates(args.User).Position;
-        foreach (var hit in args.HitEntities)
-        {
-            var hitPos = _transform.GetMapCoordinates(hit).Position;
-            var angle = args.Direction ?? hitPos - mapPos;
-            if (angle == Vector2.Zero)
-                continue;
-
-            if (!CanThrowOnHit(ent, hit))
-                continue;
-
-            if (comp.UnanchorOnHit && HasComp<AnchorableComponent>(hit))
-            {
-                _transform.Unanchor(hit, Transform(hit));
-            }
+        if (_delay.IsDelayed(weapon.Owner))
+            return;
```

Comment:
This should be cancelling the entire melee hit I would think.

Reply:
Wouldn't that be handled in the melee system? This isn't really like "if use delay is active don't melee" it's "if use delay is active don't throw this person"

Reply:
Having the melee swing decoupled from the melee effect is bizarre I would think? Especially when it's using generic usedelays (which are standardised to around 1 - 1.5s) and not its own usedelay.

Reply:
I still think it's different, and we don't really have a good way to display effect use delays (like this) vs what's on an item already.

For example if UseDelay is on a bat, what is it used for? Wielding uses the Bat's UseDelay to show that there's a use delay in effect. And in that effect use delay still doesn't prevent you from attacking, unless the "requires wield" was on the bat as well.

And if it wasn't on the bat and instead was on the comp, how do you effectively:
1) Show that this is the usedelay coming from a different comp
2) Tell when a usedelay should be applying towards something.

That being said I think that's completely out of scope for this.